frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -49,43 +51,48 @@ const selectorReducer = (state: State, action: Action): State => {
     case "ADD_PROJECTS": {
       let newState = { ...state };
 
-      // a given custom project may already exist in the group so don't trigger a state update for those duplicates
+      // a given custom project may already exist in the group so don't trigger a state update for the duplicate(s)
       const uniqueCustomProjects = action.payload.projects.filter(
-        (project: string) => !(project in newState[action.payload.group])
+        (project: string) => !(project in state[action.payload.group])
       );
       if (uniqueCustomProjects.length === 0) {
-        return newState;
+        // since there's no new projects to add, we return the original state as it wasn't modified
+        return state;
```

Comment:
per comment https://github.com/lyft/clutch/pull/1571#discussion_r662440427, I also changed this part to return the original state